snakemake/scheduler.py

```diff
@@ -647,7 +656,9 @@ def job_selector_ilp(self, jobs):
                     scheduled_jobs[job] * self.required_by_job(temp_file, job)
                     for job in jobs
                 ]
-            ) / lpSum([self.required_by_job(temp_file, job) for job in jobs])
+            ) / lpSum(
+                [self.required_by_job(temp_file, job) for job in self.remaining_jobs]
```

Comment:
self.remaining_jobs into a variable above, such that it does not need to be recomputed for each constraint.